hack/make-rules/test.sh

```diff
@@ -211,6 +211,38 @@ junitFilenamePrefix() {
   echo "${KUBE_JUNIT_REPORT_DIR}/junit_${KUBE_TEST_API_HASH}_$(kube::util::sortable_date)"
 }
 
+verifyAndSuggestPackagePath() {
+  local specified_package_path="$1"
+  local alternative_package_path="$2"
+  local original_package_path="$3"
+  local suggestion_package_path="$4"
+
+  if ! [ -d "$specified_package_path" ]; then
+    kube::log::error "specified test path \"$specified_package_path\" does not exist"
+
+    if [ -d "$alternative_package_path" ]; then
+      kube::log::info "if you intended to test \"$alternative_package_path\", change
+      \"$original_package_path\" to \"$suggestion_package_path\""
+    fi
+    exit 1
+  fi
+}
+
+verifyPathsToPackagesUnderTest() {
+  local packages_under_test=($@)
+
+  for package_path in "${packages_under_test[@]}"; do
+    local local_package_path="$(pwd)/$package_path"
+    local go_package_path="$GOPATH/src/$package_path"
+
+    if [[ "${package_path:0:2}" == "./" ]] ; then
+      verifyAndSuggestPackagePath "$local_package_path" "$go_package_path" "$package_path" "${package_path:2}"
```

Comment:
what's `"${package_path:2}"`?

Reply:
oh, ic

Reply:
The error messages are a bit confusing though because the test framework works under it's own gopath. Have a look at this error when I intentionally specify the wrong path:
```
$ make test WHAT=./k8s.io/kubernetes/pkg/controller/podautoscaler
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1alpha1,admission.k8s.io/v1alpha1,apps/v1beta1,apps/v1beta2,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,,federation/v1beta1
!!! [0925 10:22:23] specified test path "/opt/k8sgopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/./k8s.io/kubernetes/pkg/controller/podautoscaler" does not exist
if you intended to test "/opt/k8sgopath/src/k8s.io/kubernetes/_output/local/go/src/./k8s.io/kubernetes/pkg/controller/podautoscaler", change
      "./k8s.io/kubernetes/pkg/controller/podautoscaler" to "k8s.io/kubernetes/pkg/controller/podautoscaler"
make: *** [test] Error 1
```

Reply:
Great point @cblecker ! Updated the pr with clearer error messages.

```
$ make test WHAT=pkg/controller/podautoscaler`
specified test path '$GOPATH/src/pkg/controller/podautoscaler' does not exist
try changing "pkg/controller/podautoscaler" to "./pkg/controller/podautoscaler"
```
and
```
$ make test WHAT=./k8s.io/kubernetes/pkg/controller/podautoscaler
specified test path './k8s.io/kubernetes/pkg/controller/podautoscaler' does not exist.
try changing './k8s.io/kubernetes/pkg/controller/podautoscaler' to 'k8s.io/kubernetes/pkg/controller/podautoscaler'
```

Do you think that's easier to parse?